client/app/components/ParameterApplyButton.jsx

```diff
@@ -7,16 +7,22 @@ import KeyboardShortcuts from "@/services/KeyboardShortcuts";
 
 function ParameterApplyButton({ paramCount, onClick }) {
   // show spinner when count is empty so the fade out is consistent
-  const icon = !paramCount ? "spinner fa-pulse" : "check";
+  const icon = !paramCount ? (
+    <span role="status" aria-live="polite" aria-relevant="additions removals">
+      <i className="fa fa-spinner fa-pulse" aria-hidden="true" />
+      <span className="sr-only">Loading...</span>
+    </span>
+  ) : (
+    <i className="fa fa-check" aria-hidden="true" />
+  );
 
   return (
     <div className="parameter-apply-button" data-show={!!paramCount} data-test="ParameterApplyButton">
+      {/* TODO: This button stays detectable even when invisible */}
```

Comment:
Same here, but I'd actually favor just creating Follow Up issues to track these instead of putting `TODO`s in the application (they are usually forsaken).

Reply:
I've been doing both, particularly because I intend to tackle these in the near future and it helps me keeping it in mind. Are you in favor of not keeping the comments for a particular reason?

Reply:
My reason is that they bring little or no value. Even more when it is about new features as they always have a "where to be improved". The result is that we start to have multiple comments in the codebase that are not relevant to who's reading the code as they won't even be aware of the context 🤷‍♂️.

That said, I'm against TODO comments in the code. There are a couple of exceptions I consider for them however:
1. When you actually want to index your files with some easily searchable term (e.g.: we used `ANGULAR_REMOVE_ME` to track any of the still dependent leftovers while migrating away from it)
2. When the `TODO` is actually relevant to anyone that will read the code (I can't think about an example for this now 😕)

Anyway, this is more of an opinion, maybe someone else has a different thought...

Reply:
SGTM!